controllers/imageupdateautomation_controller.go

```diff
@@ -265,54 +261,39 @@ func (r *ImageUpdateAutomationReconciler) Reconcile(ctx context.Context, req ctr
 
 	debuglog.Info("attempting to clone git repository", "gitrepository", originName, "ref", ref, "working", tmp)
 
-	access, err := r.getRepoAccess(ctx, &origin)
+	authOpts, err := r.getAuthOpts(ctx, &origin)
 	if err != nil {
 		return failWithError(err)
 	}
 
-	// We set the TransportOptionsURL of this set of authentication options here by constructing
-	// a unique URL that won't clash in a multi tenant environment. This unique URL is used by
-	// libgit2 managed transports. This enables us to bypass the inbuilt credentials callback in
-	// libgit2, which is inflexible and unstable.
-	// NB: The Transport Options URL must be unique, therefore it must use the object under
-	// reconciliation details, instead of the repository it depends on.
-	if strings.HasPrefix(origin.Spec.URL, "http") {
-		access.auth.TransportOptionsURL = fmt.Sprintf("http://%s/%s/%d", auto.Name, auto.UID, auto.Generation)
-	} else if strings.HasPrefix(origin.Spec.URL, "ssh") {
-		access.auth.TransportOptionsURL = fmt.Sprintf("ssh://%s/%s/%d", auto.Name, auto.UID, auto.Generation)
-	} else {
-		return failWithError(fmt.Errorf("git repository URL '%s' has invalid transport type, supported types are: http, https, ssh", origin.Spec.URL))
+	lgc, err := libgit2pkg.NewClient(tmp, authOpts)
+	if err != nil {
+		return failWithError(err)
+	}
+	defer lgc.Close()
+
+	opts := git.CloneOptions{}
+	if ref != nil {
+		opts.Tag = ref.Tag
+		opts.SemVer = ref.SemVer
+		opts.Commit = ref.Commit
+		opts.Branch = ref.Branch
 	}
```

Comment:
This could be a helper func, as we will need to convert ref to `CloneOptions` here and in tests.